src/daft-scan/src/scan_task_iters.rs

```diff
@@ -308,3 +309,40 @@ pub(crate) fn split_by_row_groups(
         )
     }
 }
+
+fn split_and_merge_pass(
+    scan_tasks: Arc<Vec<ScanTaskLikeRef>>,
+    pushdowns: &Pushdowns,
+    cfg: &DaftExecutionConfig,
+) -> DaftResult<Arc<Vec<ScanTaskLikeRef>>> {
+    // Perform scan task splitting and merging if there are only ScanTasks (i.e. no DummyScanTasks).
+    if scan_tasks.iter().all(|st| st.is_scan_task()) {
```

Comment:
```suggestion
    if scan_tasks.iter().all(|st| st. as_any_arc().downcast::<ScanTask>().is_ok()) {
```

Reply:
and also delete `is_scan_task()` from trait def

Reply:
Done!